lib/mirage.ml

```diff
@@ -669,12 +669,18 @@ let opt_map f = function Some x -> Some (f x) | None -> None
 let (@?) x l = match x with Some s -> s :: l | None -> l
 let (@??) x y = opt_map Key.abstract x @? y
 
+(* convenience function for linking tcpip.unix or .xen for checksums *)
+let right_tcpip_library ?min ?max ?ocamlfind ~sublibs pkg =
+  Key.(if_ is_unix)
+    [ package ?min ?max ?ocamlfind ~sublibs:("unix"::sublibs) pkg ]
+    [ package ?min ?max ?ocamlfind ~sublibs:("xen"::sublibs) pkg ]
```

Comment:
yes, it's slightly awkward how different `backends` do this (atm -- actually since several years).  the simple approach to contain (most!) bindings in a single package (as done by solo5) seems to be more convenient atm, but a real cross-compilation story is needed...